File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution
 {
    void f(int ind,int[] nums,int target,List<List<Integer>>ans,  List<Integer>list)
    {
        if(target==0)
        {
            // ans.add(list);
               ans.add(new ArrayList<>(list));
            return;
        }
        if(ind>=nums.length || target<0)
        return;
        for(int i=ind;i<nums.length;i++)
        {
            if(nums[i]<=target)
            {
                list.add(nums[i]);
                f(i,nums,target-nums[i],ans,list);
                list.remove(list.size()-1);
            }
        }
    }
    public List<List<Integer>> combinationSum(int[] nums, int target)
     {
        List<List<Integer>>ans=new ArrayList<>();
        List<Integer>list=new ArrayList<>();
        f(0,nums,target,ans,list);
        return ans;
    }
}

File: Data Structures & Algorithms/next-greater-element-i/submission-0.cpp
class Solution
 {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2)
     {
        
          int n1=nums1.size();
        int n2=nums2.size();
        // int [] ans=new int[n1];
        vector<int>ans(n1);
        for(int i=0;i<n1;i++)
        {
            ans[i]=-1;
            for(int j=0;j<n2;j++)
            {
                  if(nums1[i]==nums2[j])
                {
                for(int k=j+1;k<n2;k++)
                {
              
                   if(nums2[k]>nums2[j])
                   {
                    ans[i]=nums2[k];
                    break;

                   }
               
                }
                break;
                }
            }
        }
        return ans;
    }
};